src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/custom.py

```diff
@@ -0,0 +1,167 @@
+                                        defaulted subnet this will be set to true or false
+    :param artifacts: JSON encoded array of artifacts to be applied. Use @{file} to load from a file
+    :param location: Location where to create VM. Defaults to the location of the lab
+    :param tags: Space separated tags in 'key[=value]' format. Use "" to clear existing tags
+    :param custom_image_id:
+    :param is_authentication_with_ssh_key:
+    :param lab_virtual_network_id:
+    :param os_publisher:
+    :param os_offer:
+    :param os_sku:
+    :param os_version:
+    :param os_type:
+    :param gallery_image_reference:
+    :param generate_ssh_keys:
+    :return:
+    """
+    from azure.mgmt.devtestlabs.models.lab_virtual_machine import LabVirtualMachine
+
+    is_authentication_with_ssh_key = True if authentication_type == 'ssh' else False
+
+    lab_virtual_machine = LabVirtualMachine(notes=notes,
+                                            custom_image_id=custom_image_id,
+                                            size=size,
+                                            user_name=admin_username,
+                                            password=admin_password,
+                                            ssh_key=ssh_key,
+                                            is_authentication_with_ssh_key=is_authentication_with_ssh_key,
+                                            lab_subnet_name=subnet,
+                                            lab_virtual_network_id=lab_virtual_network_id,
+                                            disallow_public_ip_address=disallow_public_ip_address,
+                                            artifacts=artifacts,
+                                            gallery_image_reference=gallery_image_reference,
+                                            name=name,
+                                            location=location,
+                                            tags=tags)
+
+    return client.create_environment(resource_group, lab_name, lab_virtual_machine)
+
+
+def list_vm(client, lab_name, resource_group, order_by=None, top=None,
+            filters=None, my_vms=None, my_claimable_vms=None):
+    """
+    :param client: Virtual machine client
+    :param lab_name: Name of the lab
+    :param resource_group: Name of the lab resource group
+    :param order_by: The ordering expression for the results, using OData notation
+    :param top: The maximum number of resources to return from the operation
+    :param filters: The filter to apply on the operation
+    :param my_vms: List of viewable labs virtual machines. Cannot be used with --filters
+    :param my_claimable_vms: List of claimable labs virtual machines. Cannot be used with --filters
+    :return:
+    """
+
+    collection = [filters, my_vms, my_claimable_vms]
+    if not _single(collection):
+        raise CLIError("usage error: [--filters FILTER | --my-vms | --my-claimable-vms]")
```

Comment:
This should be in a validator.... but what is "vm-vms" and "my-claimable-vms"?  That's a very strange name for an argument... If they are mutually exclusive, then filter should be in the same argument group. Also, the argument group shouldn't be "OData Filters" considering you don't provide any OData anything if you use the flags. :-P